content/case-studies/nanoframework-usage-at-csa-engineering-ag.md

```diff
@@ -0,0 +1,51 @@
+
+## Application 2: Control and communication electronic for an exhibition eye-catcher product
+This product is an 8x8x8 LED-Cube to display text and animations and thus catch the attention of exhibition visitors. The LED-Cubes can be configured, animations chosen, and the status supervised by the operator on a frontend. In the background our IoT-Backend based on Azure products is running. The communication with this IoT-Backend via MQTT, the conversion and storage of the animations / playlists and the activation of the LEDs via SPI is realized with the M4MCU-Board, .NET nanoFramework as basis and the C# product application.
+
+## M4MCU-Board
+For both applications some key requirements were important:
+-	Extensive communication capabilities, especially TCP/IP for the remote access and CAN as well as ModBus for connecting to a PLC and industrial sensors
+-	Available filesystem and capabilities to handle larger data files and enable an easy on-site configuration, especially required are USB host and SD card functionalities
+-	Usage of C# code with all the publicly available .NET class libraries, as already a large code basis is existent at the customer and CSA. Additionally, to facilitate an efficient development environment as the same tools for programming the embedded part (M4MCU) and the computer application can be used
+-	The possibility of safe and secured remote software updates in the field
+
+The developed M4MCU-Board fulfils these key requirements with the following components:
+-	Cortex-M4 STM32F427 microcontroller with 256kB SRAM and 1MB flash
+-	8MB SDRAM and 8MB flash external storage
+-	SD card slot to store configuration data and large files
+-	LEDs, buttons and some GPIOs for general interactions
+-	Communication interfaces:
+  -	1x USB device
+  - 1x USB host
+  - 1x Ethernet RJ45
+  - 1x CAN
+  - 1x RS485
+
+![M4MCU-Board](../../images/case-studies/csa-engineering-ag/M4MCU.jpg)
+
+In addition, several software realized features are available as e.g., development, deploying and debugging directly with Visual Studio, data encryption with NaCl algorithm, remote software update, several communication protocols as SSL/TLS, CAN, ModBus, software supervision with a watchdog.
+
+The porting of the .NET nanoFramework to our own hardware platform took some time and required in-depth knowledge in embedded software development. But finally, the .NET nanoFramework could be successfully introduced and provided a comfortable way of development combined with a comprehensive set of features for the product application.
+
+## Images
+
+![M4MCU DIN rail case](../../images/case-studies/csa-engineering-ag/M4MCU_Housing.PNG)
+*M4MCU built into DIN rail compatible case*
+
+![M4MCU in application 1](../../images/case-studies/csa-engineering-ag/M4MCU_SmartGateway.PNG)
```

Comment:
you should include the pictures in the different sections, I feel it will be a better nice way of looking at all this